api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1818,6 +1819,24 @@ type KMSSpec struct {
 	// AWS defines metadata about the configuration of the AWS KMS Secret Encryption provider
 	// +optional
 	AWS *AWSKMSSpec `json:"aws,omitempty"`
+	// Azure defines metadata about the configuration of the Azure KMS Secret Encryption provider using Azure key vault
+	// +optional
+	Azure *AzureKMSSpec `json:"azure,omitempty"`
+}
+
+// AzureKMSSpec defines metadata about the configuration of the Azure KMS Secret Encryption provider using Azure key vault
+type AzureKMSSpec struct {
+	// Location contains the Azure region
+	// +optional
+	Location string `json:"location"`
+	// KeyVaultName is the name of the keyvault. Must match criteria specified at https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name
+	// Your Microsoft Entra application used to create the cluster must be authorized to access this keyvault, e.g using the AzureCLI:
+	// `az keyvault set-policy -n $KEYVAULT_NAME --key-permissions decrypt encrypt --spn <YOUR APPLICATION CLIENT ID>`
+	KeyVaultName string `json:"keyVaultName"`
```

Comment:
Should we go ahead and add the CEL to enforce the requirements from https://learn.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name?

Reply:
yeah I like that idea.

Reply:
Azure CLI/Console wouldn't allow you to create a vault with a name that violates the requirements in the first place, so I don't think we need to enforce that here as well, since we are not the ones creating the resources.

Reply:
Ok that makes sense